lib/wrapbox/runner/ecs.rb

```diff
@@ -561,6 +561,8 @@ class Cli < Thor
         method_option :execution_retry, type: :numeric
         method_option :max_retry_interval, type: :numeric
         method_option :ignore_signal, type: :boolean, default: false, desc: "Even if receive a signal (like TERM, INT, QUIT), ECS Tasks continue running"
+        method_option :tags
```

Comment:
`type: array` is better because we can include some commas in each tag. We are sometimes confused when `--environments IDS=1,2,3` doesn't work expectedly.

Reply:
I think thor's array type has a positional restriction.
Thor cannot detect whether a value is a part of an array or a  subcommand. (Maybe so)
It is a bit troublesome.
Repeatable String type is ideal for me.

Reply:
```ruby
require "thor"

module Sample
  module Ecs
    class Cli < Thor
      namespace :ecs

      desc "run_cmd [shell command]", "Run shell on ECS"
      method_option :config, aliases: "-f", required: true, banner: "YAML_FILE", desc: "yaml file path"
      method_option :cpu, type: :numeric
      method_option :environments, type: :array, aliases: "-e"
      method_option :tags, type: :hash
      def run_cmd(*args)
        p options
        environments = options[:environments].map{|key_value| key_value.split("=", 2) }.to_h
        p environments
        p args
      end
    end
  end

  class Cli < Thor
    register(::Sample::Ecs::Cli, "ecs", "ecs [COMMAND]", "Commands for ECS")
  end
end


Sample::Cli.start
```

```
$ bundle exec ruby sample.rb ecs run_cmd --config a.yml --cpu 1024 -e a=1 b=2 c=3,2,1 echo test
--- run_cmd ---
{"config"=>"a.yml", "cpu"=>1024, "environments"=>["a=1", "b=2", "c=3,2,1", "echo", "test"]}
Traceback (most recent call last):
        11: from sample.rb:31:in `<main>'
        10: from /home/kenji/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor/base.rb:485:in `start'
         9: from /home/kenji/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
         8: from /home/kenji/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command
'
         7: from /home/kenji/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
         6: from /home/kenji/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor.rb:243:in `block in subcommand'
         5: from /home/kenji/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor/invocation.rb:116:in `invoke'
         4: from /home/kenji/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
         3: from /home/kenji/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command
'
         2: from /home/kenji/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
         1: from sample.rb:17:in `run_cmd'
sample.rb:17:in `to_h': wrong array length at 3 (expected 2, was 1) (ArgumentError)
```

@joker1007 Do you mean the above example behavior?

Reply:
If we use `:hash` for `--environments` and `--tags`, we can use wrapbox command like following:

```
$ wrapbox ecs run_cmd -e FOO:value IDS=1,2,3 --tags key1:value1 key2:value2 echo test
```

But we cannot change the delimiter `:` to other characters such as `=`.

Reply:
> Do you mean the above example behavior?

Yes. I mean so.

Reply:
> I think thor's array type has a positional restriction.
> Repeatable String type is ideal for me.

I see. The option was introduced by https://github.com/erikhuda/thor/pull/674 and it is available since [v1.0.0](https://github.com/erikhuda/thor/releases/tag/v1.0.0), which is released about a year ago.

```ruby
require "thor"

class Cli < Thor
  desc "run_cmd [shell command]", "Run shell on ECS"
  method_option :environments, type: :string, aliases: "-e", repeatable: true
  def run_cmd(*args)
    p options
    environments = options[:environments].map{|key_value| key_value.split("=", 2) }.to_h
    p environments
    p args
  end
end
Cli.start
```

```
ruby sample.rb run_cmd -e a=1 -e b=2 -e c=3,2,1 echo test
{"environments"=>["a=1", "b=2", "c=3,2,1"]}
{"a"=>"1", "b"=>"2", "c"=>"3,2,1"}
["echo", "test"]
```

Why don't we use the option "repeatable" at least for the new method option "tags"?

Reply:
Oh, I didn't know the repeatable string option. Changed in  9d853ec.

Reply:
I think wrapbox.gemspec must also be modified:

```diff
diff --git a/wrapbox.gemspec b/wrapbox.gemspec
index 3bdd323..7e78321 100644
--- a/wrapbox.gemspec
+++ b/wrapbox.gemspec
@@ -26,7 +26,7 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency "activesupport", ">= 4"
   spec.add_runtime_dependency "docker-api"
   spec.add_runtime_dependency "multi_json"
-  spec.add_runtime_dependency "thor"
+  spec.add_runtime_dependency "thor", ">= 1"
   spec.add_runtime_dependency "thwait"

   spec.add_development_dependency "bundler"
```